src/sections/dataset/publish-dataset/PublishDatasetModal.tsx

```diff
@@ -0,0 +1,113 @@
+import { Button, Modal } from '@iqss/dataverse-design-system'
+import { SubmissionStatus } from '../../create-dataset/useCreateDatasetForm'
+import { useTranslation } from 'react-i18next'
+import type { DatasetRepository } from '../../../dataset/domain/repositories/DatasetRepository'
+import { usePublishDataset } from './usePublishDataset'
+import { VersionUpdateType } from '../../../dataset/domain/models/VersionUpdateType'
+import { Form } from '@iqss/dataverse-design-system'
+import { useState } from 'react'
+import { useSession } from '../../session/SessionContext'
+import { PublishDatasetHelpText } from './PublishDatasetHelpText'
+import { License } from '../dataset-summary/License'
+import { defaultLicense } from '../../../dataset/domain/models/Dataset'
+
+interface PublishDatasetModalProps {
+  show: boolean
+  repository: DatasetRepository
+  persistentId: string
+  releasedVersionExists: boolean
+  handleClose: () => void
+}
+
+export function PublishDatasetModal({
+  show,
+  repository,
+  persistentId,
+  releasedVersionExists,
+  handleClose
+}: PublishDatasetModalProps) {
+  const { t } = useTranslation('publishDataset')
+  const { user } = useSession()
+  const onPublishErrorCallback = () => {
+    // TODO: Navigate to error page
+    throw new Error('Error publishing dataset')
```

Comment:
For the error handling, I can see the advantage of displaying an error message in the modal, but I was trying to avoid having error handling logic all over the application, especially for unexpected runtime errors. For unexpected runtime errors, when is it recommended to go to an error page vs displaying a message in the current page?

Reply:
Exactly you can make it a named function. 
Instead of  `const onPublishErrorCallback = () => {}`
Do `function onPublishErrorCallback() {}`
And you will be able to move it to the bottom and still be able to use it before declaration.

Reply:
Yes I understand about the error handling everywhere, it gets a little bit tedious also.
In my experience if an error ocurred, its better to show some small error message to the user that something went wrong and let him retry again quickly instead of navigating to an error page.
I think you can just read the `publishError` from `usePublishDataset` hook and show it if exists.
You dont need the `onPublishErrorCallback` in this case I believe.

Reply:
ok, I added an error message to the modal 👍